### PR TITLE
Ft circleci codeclimate 187300136

### PR DESCRIPTION
**Pull request by Mukunzijames**
issue in referencing?
Questions:

What does this PR do?
configuring circleci burges in readme and codeclimate.

Description of Task to be completed?

setting up circleci and config.yml
setting up coverall.io and coverage
How should this be manually tested?
npm run dev
npm run test
Any background context you want to provide?
-dummy data are provided before having the real data

What are the relevant pivotal tracker/Trello stories?
id of the project
#187300135
#187300134
#187300136

### CODE REVIEW

**Review by IRASUBIZA-Elyse — APPROVED** [review]

File: test/server-run.js
Comment (fab-ryan) [inline test/server-run.js:1]: remove this file

File: src/database/db.ts [hunk 1]
```diff
@@ -1,28 +1,26 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
 import { Sequelize } from 'sequelize';
 import getDatabaseConfig from '../config/config';
-import Models from './models';
+import initializeModels from './models';
```
Comment (fab-ryan) [inline src/database/db.ts]: Please don't ever change someone else's code it's prohibited

**Review by fab-ryan — CHANGES_REQUESTED** [review]
work on those comments

**Review by RWEMAREMY — APPROVED** [review]
Great work

File: test/server-run.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const mocha_1 = require("mocha");
+const app_1 = __importDefault(require("../src/app"));
+chai_1.default.use(chai_http_1.default);
+const { expect } = chai_1.default;
+(0, mocha_1.describe)('API Tests', () => {
+    (0, mocha_1.it)('should return Welcome to the API!👋🏽👋🏽', done => {
+        chai_1.default
+            .request(app_1.default)
+            .get('/')
+            .end((err, res) => {
```
Comment (fab-ryan) [inline test/server-run.js]: Please remove this file

**Review by fab-ryan — CHANGES_REQUESTED** [review]
remove that file of js in test

**Review by Ruthuwamahoro — APPROVED** [review]
Great job

File: src/database/db.ts [hunk 3]
```diff
@@ -1,28 +1,26 @@
...
 const models = Models(sequelize);
 
-Object.keys(models).forEach(key => {
-  // @ts-expect-error ignore expected errors
-  if (models[key].associate) {
-    // @ts-expect-error ignore expected errors
-    models[key].associate(models);
+Object.values(models).forEach((model) => {
+  if (model.associate) {
+    model.associate(models);
```
Comment (fab-ryan) [inline src/database/db.ts:23]: this is wrong each modal can'thave associate but key on models

**Review by fab-ryan — APPROVED** [review]

**Pull request merged**